staging/src/k8s.io/apimachinery/pkg/util/proxy/transport_test.go

```diff
@@ -274,3 +275,34 @@ func TestProxyTransport(t *testing.T) {
 		testItem(name, &item)
 	}
 }
+
+func TestRewriteResponse(t *testing.T) {
+	testTransport := &Transport{
+		Scheme:      "http",
+		Host:        "foo.com",
+		PathPrepend: "/proxy/node/node1:10250",
+	}
+	encodeType := []string{"gzip", "deflate"}
+	resp := &http.Response{
+		Body: ioutil.NopCloser(bytes.NewBufferString("Unit test")),
+	}
+	errFn := func(encode string, err error) {
+		t.Errorf("%s failed to read and write: %v", encode, err)
+	}
+	for _, v := range encodeType {
+		request, err := http.NewRequest("GET", "http://mynode.com/", nil)
+		request.Header.Set("Content-Encoding", v)
+		request.Header.Add("Accept-Encoding", v)
+
+		gotResponse, err := testTransport.rewriteURL(request, resp)
+
+		if err != nil {
+			errFn(v, err)
+		}
+
+		if getResponse.Body == nil {
```

Comment:
this isn't actually testing the encoding worked correctly

Reply:
@liggitt Updated. PTAL
Thanks!!